From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Sat, 13 Jun 2020 22:25:54 +0200
Subject: Fix format specifiers for 32bit
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

client-authenticate.c: In function ‘cmd_helo_reply’:
client-authenticate.c:64:13: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 4 has type ‘uoff_t’ {aka ‘const long long unsigned int’} [-Wformat=]
   64 |     "SIZE", "%"PRIuSIZE_T,
      |             ^~~
   65 |     subm_client->set->submission_max_mail_size);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                     |
      |                     uoff_t {aka const long long unsigned int}
In file included from ../../src/lib/lib.h:6,
                 from ../../src/login-common/login-common.h:4,
                 from client-authenticate.c:3:
../../config.h:762:21: note: format string is defined here
  762 | #define PRIuSIZE_T "u"

cmd-havespace.c: In function ‘cmd_havespace’:
cmd-havespace.c:55:22: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 6 has type ‘uoff_t’ {aka ‘long long unsigned int’} [-Wformat=]
   55 |  e_debug(e->event(), "Quota is within limits for script `%s' "
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |   "with size %"PRIuSIZE_T, scriptname, size);
      |                                        ~~~~
      |                                        |
      |                                        uoff_t {aka long long unsigned int}
/<<PKGBUILDDIR>>/src/lib/event-log.h:72:43: note: in definition of macro ‘e_debug’
   72 |   e_debug(_tmp_event, __FILE__, __LINE__, __VA_ARGS__); \
      |                                           ^~~~~~~~~~~
In file included from /<<PKGBUILDDIR>>/src/lib/lib.h:6,
                 from cmd-havespace.c:4:
/<<PKGBUILDDIR>>/config.h:762:21: note: format string is defined here
  762 | #define PRIuSIZE_T "u"
---
 pigeonhole/src/managesieve/cmd-havespace.c | 2 +-
 src/submission-login/client-authenticate.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pigeonhole/src/managesieve/cmd-havespace.c b/pigeonhole/src/managesieve/cmd-havespace.c
index 468a923..5af2bb1 100644
--- a/pigeonhole/src/managesieve/cmd-havespace.c
+++ b/pigeonhole/src/managesieve/cmd-havespace.c
@@ -53,7 +53,7 @@ bool cmd_havespace(struct client_command_context *cmd)
 	struct event_passthrough *e =
 		client_command_create_finish_event(cmd);
 	e_debug(e->event(), "Quota is within limits for script `%s' "
-		"with size %"PRIuSIZE_T, scriptname, size);
+		"with size %"PRIuUOFF_T, scriptname, size);
 
 	client_send_ok(client, "Putscript would succeed.");
 	return TRUE;
diff --git a/src/submission-login/client-authenticate.c b/src/submission-login/client-authenticate.c
index 3b2680a..0648f54 100644
--- a/src/submission-login/client-authenticate.c
+++ b/src/submission-login/client-authenticate.c
@@ -61,7 +61,7 @@ static void cmd_helo_reply(struct submission_client *subm_client,
 
 		if (subm_client->set->submission_max_mail_size > 0) {
 			smtp_server_reply_ehlo_add_param(reply,
-				"SIZE", "%"PRIuSIZE_T,
+				"SIZE", "%"PRIuUOFF_T,
 				subm_client->set->submission_max_mail_size);
 		} else {
 			smtp_server_reply_ehlo_add(reply, "SIZE");
